cpp/benchmarks/io/cuio_common.cpp

```diff
@@ -141,3 +141,16 @@ std::vector<cudf::size_type> segments_in_chunk(int num_segments, int num_chunks,
 
   return selected_segments;
 }
+
+void drop_cache()
+{
+  if (std::getenv("CUDF_BENCHMARK_DROP_CACHE") != nullptr) {
```

Comment:
I find it confusing that a function called `drop_cache` _may_ or may not drop the cache. This would become especially confusing if there were cases where we always needed to drop the cache. I would recommend separating this into two functions like `drop_cache` and `maybe_drop_cache` where the latter only calls the former if the environment variable is set. Since this is just a benchmarking tool and not publicly exposed it's not a big deal and maybe not worth overengineering, so I'll let you decide if you think that abstraction is worthwhile.

Reply:
I don't think we ever could write a programattic way to 100% drop the cache. Naming the function something like `try_to_drop_cache` most likely better captures the intent.

Reply:
renamed to `try_drop_l3_cache`, please resolve if this naming is fine.

Reply:
I'm happy with the naming change. I just want it clear what the function is doing, it's fine to not generalize to other caches etc.